keylime/tpm/tpm_abstract.py

```diff
@@ -230,21 +230,37 @@ def __check_ima(self, agent_id, pcrval, ima_measurement_list, allowlist, ima_key
         logger.debug(f"IMA measurement list of agent {agent_id} validated")
         return True
 
-    def check_pcrs(self, agent_id, tpm_policy, pcrs, data, virtual, ima_measurement_list, allowlist, ima_keyring, mb_measurement_list, mb_intended_state):
+    def check_pcrs(self, agent_id, tpm_policy, pcrs, data, virtual, ima_measurement_list, allowlist, ima_keyring, mb_measurement_list, mb_refstate):
         try:
             tpm_policy_ = ast.literal_eval(tpm_policy)
         except ValueError:
             tpm_policy_ = {}
         pcr_allowlist = tpm_policy_.copy()
 
-        if mb_measurement_list or mb_intended_state :
+        if mb_measurement_list or mb_refstate :
             logger.info("Measured boot information received, but for now it will not be processed. A future update will enable the full processing of it.")
 
         if 'mask' in pcr_allowlist:
             del pcr_allowlist['mask']
         # convert all pcr num keys to integers
         pcr_allowlist = {int(k): v for k, v in list(pcr_allowlist.items())}
 
+        if mb_refstate not in (False, None) and mb_measurement_list not in (False, None) :
+            mb_measurement_data = self.parse_bootlog(mb_measurement_list)
+            if not isinstance(mb_measurement_data, dict):
```

Comment:
What scenarios exist where this variable wouldn't be a dict?

Reply:
Ah, this is a bit of defensive programming: intel's tpm2-tools (still used, even without all the required features) might output badly formed data (with an exit code 0 mind you) which will be misinterpreted. We are checking if it is indeed a dict in order to detect failure.

Reply:
That makes sense for the things inside of `mb_measurement_data` (like the 'pcrs' key used later). But `mb_measurement_data` is created by `parse_bootlog` which eventually calls `config.yaml_to_dict` and there can't be a case where `yaml_to_dict` where it returns a non-None value that isn't a dict, right? Also, if we need to do defensive programming on what `yaml_to_dict` could return that isn't a dict it should probably be done where `yaml_to_dict` is itself called.

Reply:
Yes, this particular test is an overreach. By way of excuses, I have just this: we have discussed ad nauseam how to *not trust* anything coming from the agent. Eventually the whole parsing process will be done in tpm2_eventlog and the code will slim down a lot. 

Hai Huang is spending a lot of time with the Intel guys doing the first line of defense in tpm2_eventlog: if a malicious agent submits a malformed quote tpm2_eventlog could crash or output garbage.

Reply:
Again, if we are not trusting the external tool, it is not given that `yaml_to_dict` returns a dict.  All we know is that it returns something that `yaml.load` can return.

Reply:
If there's a possibility that we're giving data to `yaml.load` inside of `yaml_to_dict` that could not be a dict, then that needs to be handled in `yaml_to_dict` not everywhere it could be used. So I agree that if we're not trusting it we should be defensive, just not about sprinkling it all over the place, but centralized.